datasetinsights/datasets/synthetic.py

```diff
@@ -137,7 +156,30 @@ def __init__(
             random_seed (int): random seed used for splitting dataset into
                 train and val
         """
-        self.dataset_directory = data_path
+        # check if dataset files are present
+        if SynDetection2D.is_synthetic_dataset_files_present(data_path):
```

Comment:
Right now, pulbic synthdet dataset as well as groceries real dataset files are zipped into single zip file. And HTTP downloader unzips all files directly to `data_path`. Why do we need to unzip them into `unzip_folder` because then dataset class need to know unzip folder for each dataset, on the other hand Dataset class just look in `data_path` for filed and don't need to know anything about `unzipped_folder`.